app/models/miq_request/purging.rb

```diff
@@ -0,0 +1,27 @@
+class MiqRequest
+  module Purging
+    extend ActiveSupport::Concern
+    include PurgingMixin
+
+    module ClassMethods
+      def purge_date
+        ::Settings.miq_request.history.keep_miq_requests.to_i_with_method.seconds.ago.utc
+      end
+
+      def purge_window_size
+        ::Settings.miq_request.history.purge_window_size
+      end
+
+      def purge_scope(older_than)
+        MiqRequest.where(arel_table[:created_on].lt(older_than))
+      end
+
+      # this is MiqRequest#destroy hook (defined by relationships on MiqRequest)
+      def purge_associated_records(ids)
+        # sorry for the destroy. this needs to be recursive:
+        MiqRequestTask.where(:miq_request_id => ids).destroy_all
+        MiqApproval.where(:miq_request_id => ids).delete_all
```

Comment:
Should this be done in batches or does the mixin already do that?

Reply:
the miq requests are batched. So I was thinking that we would just get rid of all the associated approvals.
Since this is a `delete_all`, no records will come back over the wire so I was less concerned around this one.

(it is the `destroy_all` that is concerning to me)

Reply:
yeah I was talking about the destroy_all in batches - didn't realize I commented on the wrong line.

Reply:
I'm guessing there are probably 1 task per request. So the batching size should be about the same.
Not sure if this does it one by one.

Since we implemented `purge_associated_records`, we download all the ids.